prov/tcp/src/tcpx_conn_mgr.c

```diff
@@ -240,7 +240,9 @@ static void tcpx_cm_recv_resp(struct util_wait *wait,
 err2:
 	free(cm_entry);
 err1:
+	fastlock_acquire(&ep->lock);
```

Comment:
Thanks - we should find a way to assert that locks are being held from functions that expect them to be.  It would be of limited use here, since these changes are all in hard to hit error cases.

Reply:
Can you assert in tcpx_ep_disable that fastlock_tryacquire(&ep->lock) fails?

Reply:
Hmm... good idea.  Actually, fastlock has a debug version that carries an 'in_use' flag.  We could easily check that that's set.  We could probably extend the structure to record the thread id holding the lock.

Reply:
Btw, I'll work on adding this feature